src/Compilers/VisualBasic/Test/Semantic/SourceGeneration/GeneratorDriverTests_Attributes_SimpleName.vb

```diff
@@ -0,0 +1,1923 @@
+                                                                                              Dim input = ctx.ForAttributeWithSimpleName(Of ClassStatementSyntax)("X")
+                                                                                              ctx.RegisterSourceOutput(input, Sub(spc, node)
+                                                                                                                              End Sub)
+                                                                                          End Sub))
+
+            Dim driver As GeneratorDriver = VisualBasicGeneratorDriver.Create(ImmutableArray.Create(Of ISourceGenerator)(generator), parseOptions:=parseOptions, driverOptions:=New GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps:=True))
+            driver = driver.RunGenerators(compilation)
+            Dim runResult = driver.GetRunResult().Results(0)
+
+            Assert.Collection(runResult.TrackedSteps("result_ForAttribute"),
+                Sub(_step) Assert.True(IsClassStatementWithName(_step.Outputs.Single().Value, "C")))
+        End Sub
+
+        <Fact>
+        Public Sub FindFullAttributeOnTopLevelClass_WhenSearchingForClassDeclaration_FullAttributeName1()
+            Dim source = "
+<XAttribute>
+class C
+end class
+"
+            Dim parseOptions = TestOptions.RegularLatest
+            Dim compilation = CreateCompilation(source, options:=TestOptions.DebugDll, parseOptions:=parseOptions)
+
+            Assert.Single(compilation.SyntaxTrees)
+
+            Dim generator = New IncrementalGeneratorWrapper(New PipelineCallbackGenerator(Sub(ctx)
+                                                                                              Dim input = ctx.ForAttributeWithSimpleName(Of ClassStatementSyntax)("XAttribute")
+                                                                                              ctx.RegisterSourceOutput(input, Sub(spc, node)
+                                                                                                                              End Sub)
+                                                                                          End Sub))
+
+            Dim driver As GeneratorDriver = VisualBasicGeneratorDriver.Create(ImmutableArray.Create(Of ISourceGenerator)(generator), parseOptions:=parseOptions, driverOptions:=New GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps:=True))
+            driver = driver.RunGenerators(compilation)
+            Dim runResult = driver.GetRunResult().Results(0)
+
+            Assert.Collection(runResult.TrackedSteps("result_ForAttribute"),
+                Sub(_step) Assert.True(IsClassStatementWithName(_step.Outputs.Single().Value, "C")))
+        End Sub
+
+        <Fact>
+        Public Sub FindAttributeOnTopLevelClass_WhenSearchingForClassDeclaration_WithLocalAlias1()
```

Comment:
Think there is an opportunity here as well for a `<Theory>` based test that would eliminate a lot of duplication for the `Imports A = ...` cases.